apps/passport-client/src/validateState.ts

```diff
@@ -151,38 +142,24 @@ export function getRunningAppStateValidationErrors(
   const loggedOut = !self;
 
   // Find identity PCD in the standard way, using a known commitment.
-  let identityV3PCDFromCollection: SemaphoreIdentityPCD | undefined = undefined;
-  let identityV4PCDFromCollection: SemaphoreIdentityV4PCD | undefined =
-    undefined;
+  let identityPCDFromCollection: SemaphoreIdentityPCD | undefined = undefined;
 
   if (self && pcds) {
-    identityV3PCDFromCollection = findUserIdentityV3PCD(pcds, self);
-    identityV4PCDFromCollection = findUserIdentityV4PCD(pcds, self);
+    identityPCDFromCollection = findUserIdentityPCD(pcds, self);
   } else if (identity && identityV4 && pcds) {
```

Comment:
```suggestion
  } else if (identity && pcds) {
```
You can now look up the identity PCD with only V3 commitment, and you may as well check it even if identityV4 is missing.